packages/tools/src/tools/CursorCrosshairSync.ts

```diff
@@ -0,0 +1,581 @@
+import { AnnotationTool } from './base';
+import {
+  getEnabledElement,
+  metaData,
+  StackViewport,
+  VolumeViewport,
+  utilities,
+  Enums,
+} from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+import {
+  addAnnotation,
+  getAnnotations,
+  removeAnnotation,
+} from '../stateManagement/annotation/annotationState';
+import { isAnnotationVisible } from '../stateManagement/annotation/annotationVisibility';
+import { drawCircle as drawCircleSvg, drawLine } from '../drawingSvg';
+import { getViewportIdsWithToolToRender } from '../utilities/viewportFilters';
+import {
+  EventTypes,
+  PublicToolProps,
+  ToolProps,
+  SVGDrawingHelper,
+  Annotation,
+  Annotations,
+} from '../types';
+import { CursorCrosshairSync } from '../types/ToolSpecificAnnotationTypes';
+
+import triggerAnnotationRenderForViewportIds from '../utilities/triggerAnnotationRenderForViewportIds';
+import { StyleSpecifier } from '../types/AnnotationStyle';
+import { vec3 } from 'gl-matrix';
+
+/**
+ * CursorCrosshairSyncTool is a tool that will show your cursors position in all other elements in the toolGroup if they have a matching FrameOfReference relative to its position in world space.
+ * Also when positionSync is enabled, it will try to sync viewports so that the cursor can be displayed in the correct position in all viewports.
+ */
+class CursorCrosshairSyncTool extends AnnotationTool {
```

Comment:
So I talked internally, and we think maybe `referenceCursors` would be a better name, since it aligns with the `referenceLines` which is basically the same concept to show lines on viewports calculated based on their positioning with each other. What do you think?